Core/clim-basic/setf-star.lisp

```diff
@@ -33,37 +33,51 @@
 			    (symbol-name '#:star))
 		    (symbol-package name-sym)))))
 
-(defun make-arglist (args)
-  "Convert the list ARGS to a list of arguments in the form (var
-init-form supplied-p-parameter)."
-  (mapcar (lambda (a) `(,a nil ,(gensym))) args))
-
 (defmacro defgeneric* (fun-name lambda-list &body options)
   "Defines a SETF* generic function.  FUN-NAME is a SETF function
 name.  The last required argument is the single argument to the function in a
 SETF place form; the other arguments are values collected from the
 SETF new value form."
   (unless (setf-name-p fun-name)
     (error "~S is not a valid name for a SETF* generic function." fun-name))
-  (let ((setf-name (cadr fun-name))
-        (gf (make-setf*-gfn-name fun-name)))
-    (multiple-value-bind (required optional rest keys allow-other-keys aux keyp)
-        (alexandria:parse-ordinary-lambda-list lambda-list)
-      (when aux
-        (error "The use of &aux is not allowed in generic function lambda list."))
-      (let ((optional (make-arglist (mapcar #'car optional)))
-            (keys (make-arglist (mapcar #'cadar keys))))
+  (multiple-value-bind (required optional rest keys allow-other-keys aux keyp)
+      (alexandria:parse-ordinary-lambda-list lambda-list)
+    (when aux
+      (error "The use of &aux is not allowed in generic function lambda list."))
+    ;; See https://gitlab.com/embeddable-common-lisp/ecl/-/issues/583.
+    ;; Basically ECL and SBCL punts on defsetf keyword arguments which
+    ;; are not interned in a keyword package.
+    (loop for (key &rest rest) in keys
+          unless (keywordp (car key))
+            do (let ((new-key (alexandria:make-keyword (car key))))
+                 (warn "Coercing ~s to keyword ~s." (car key) new-key)
+                 (setf (car key) new-key)))
```

Comment:
I think we should not do anything here and just assume the implementations will get fixed.

Reply:
you are right, fixed.